### DISCUSSION TITLE
Some tests take more than 60 seconds in a standard (maybe slow) notebook

### DISCUSSION THREAD

**Discussion by oxarbitrage** [ZcashFoundation/zebra#3752] [Team Decisions]
We want to decrease the time of execution of at least the following state tests:

```
test service::finalized_state::tests::prop::blocks_with_v5_transactions has been running for over 60 seconds
test service::non_finalized_state::tests::prop::finalized_equals_pushed has been running for over 60 seconds
test service::non_finalized_state::tests::prop::forked_equals_pushed has been running for over 60 seconds
test service::non_finalized_state::tests::prop::rejection_restores_internal_state has been running for over 60 seconds
```

The default number of tests is already at 1 (https://github.com/ZcashFoundation/zebra/blob/main/zebra-state/src/service/finalized_state/tests/prop.rs#L18) so we can't try the simple fix we did at https://github.com/ZcashFoundation/zebra/pull/2574

**Comment by mpguerra** [comment]
@ZcashFoundation/zebra-team Is this still happening? do we need to do anything here? Or can we close this for now?

**Comment by oxarbitrage** [comment]
Yes this is still happening. I don't think we will fix it soon but eventually some day we should try to improve the speed of some of this tests.

**Comment by teor2345** [comment]
> Yes this is still happening. I don't think we will fix it soon but eventually some day we should try to improve the speed of some of this tests.

We try to keep most tests to 10-20 seconds, and I think these tests are within that limit on most machines.

But they're some of our slowest tests, and we really want to run more than 1 case for good test coverage. So this ticket is worth keeping.

**Comment by mpguerra** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Engineering-Team-5fb24d9264a3e8000e666a9e/issues/ZcashFoundation/zebra/2575?planning-poker) with ZenHub @conradoplg @dconnolly @jvff @oxarbitrage @teor2345 @upbqdn

**Comment by oxarbitrage** [comment]
In we added some tests that are also a bit slow, we should consider decreasing the number of random transactions the tests do.

```
test methods::tests::prop::invalid_transaction_results_in_an_error has been running for over 60 seconds
test methods::tests::prop::mempool_errors_are_forwarded has been running for over 60 seconds
test methods::tests::prop::mempool_receives_raw_transaction has been running for over 60 seconds
test methods::tests::prop::non_hexadecimal_string_results_in_an_error has been running for over 60 seconds
```

**Comment by conradoplg** [comment]
We can resurrect #3102 if needed - what blocked that is that we either

- need to generate a smaller number of shielded txs only for the chain tests, which is very awkward to implement and requires a lot of changes (or at least I couldn't find a simple way to implement it)
- use some number that is not too small so that we can keep a similar test coverage. Maybe 0.25? (i.e. 1 in 4 arbitrary txs will be shielded)

**Comment by mpguerra** [comment]
closing as stale
